src/variable.rs

```diff
@@ -329,6 +330,37 @@ impl ProblemVariables {
         Variable::at(index)
     }
 
+    /// Adds many variables with the given definitions
+    ///
+    /// ```
+    /// use good_lp::*;
+    /// // Solve a problem with 11 variables: x, y0, y1, ..., y9
+    /// variables!{problem: 2 <= x <= 3;}
+    /// let vars = vec![variable().min(0); 10];
+    /// let y: Vec<Variable> = problem.add_all(vars);
+    /// let objective: Expression = y.iter().sum(); // Minimise sum(y_i for i in [0; 9])
+    /// let mut model = problem.minimise(objective).using(default_solver);
+    /// // for all i, we must have y_i >= x
+    /// for y_i in y.iter() {
+    ///   model = model.with(constraint!(y_i >= x));
+    /// }
+    /// let solution = model.solve().unwrap();
+    /// # use float_eq::assert_float_eq;
+    /// assert_float_eq!(solution.value(y[3]), 2., abs <= 1e-8);
+    /// ```
+    pub fn add_all<T, B: FromIterator<Variable>>(&mut self, var_defs: T) -> B
+    where
+        T: IntoIterator<Item = VariableDefinition>,
+    {
+        let index = self.variables.len();
+        self.variables.extend(var_defs);
+        self.initial_count += self.variables[index..]
+            .iter()
+            .filter(|v| v.initial.is_some())
+            .count();
+        (index..self.variables.len()).map(Variable::at).collect()
```

Comment:
@lovasoa do you think we should move the `collect` call to consuming code and directly return an iterator?
```suggestion
        (index..self.variables.len()).map(Variable::at)
```
People could then do
```rs
let y: Vec<Variable> = problem.add_all(vars).collect();
```
I don't need this myself and I can't really imagine why anyone would want that, especially because we still have to make sure the iterator is fully consumed, but I'm still interested in your input.